src/shared/dissect-image.c

```diff
@@ -312,6 +317,14 @@ static int wait_for_partition_device(
         if (r < 0)
                 return r;
 
+        r = sd_device_monitor_filter_add_match_parent(monitor, parent, true);
```

Comment:
aren't non-BPF filters problematic from the performance point of view?

Reply:
In this case, it should not. As the received devices are already filtered by their subsystem and devtype.